lms/models/json_settings.py

```diff
@@ -1,14 +1,46 @@
 import base64
-from typing import Optional
+from dataclasses import dataclass
+from typing import Any, Optional, Tuple
 
 import sqlalchemy as sa
 from sqlalchemy.ext.mutable import MutableDict
 from sqlalchemy.orm import InstrumentedAttribute
 
 
-class ApplicationSettings(MutableDict):
+@dataclass
+class JSONSetting:
+    """Describe a permitted field in a JSONSettings object."""
+
+    # Helper to declare settings as secret. This can be used with format
+    AES_SECRET = object()
+
+    group: str
+    """The group name that this setting is a part of."""
+
+    key: str
+    """The key within the grouo that this setting is a part of."""
+
+    format: Any = str
```

Comment:
I reckon this one needs a bit more clarification

If understand it correctly this is both an identifier of the type  but also a callable that must take a str (`Callble[[str],str]`.

Except for AES_SECRET, which doesn't need to be a function because we exclude it in that case.

Maybe this could be solved with some extra info here on the comment or maybe we could haveL

`type: Any = str` which will take values: str|bool|AES_SECRET

and `format` or `formatted` as either a function or property that "renders" the setting accordingly to the objects own type?

Reply:
Hmm, you spotted that? This is definitely neither a type nor a formatter, but as you say, awkardly both at the same time sometimes, others not.

I was honestly trying to avoid the question for this PR. `asbool` and `str` etc. isn't quite the full story for either as we have special rules around nulls. We would probably have to create our own to handle that, and the secrets would probably always be out of scope as they'd pull in a service.

So I think nothing off the shelf will do, and if we did something manual, it might as well encompass type and formatting in one.